### PR TITLE
HttpUtil.getCharset() fails for charset in double-quotes

### LINKED ISSUE #11370: HttpUtil.getCharset() fails for charset in double-quotes
I understood from [RFC 7231](https://datatracker.ietf.org/doc/html/rfc7231#section-3.1.1.1) the content-type header can have charset value in double-quotes. 
```
media-type = type "/" subtype *( OWS ";" OWS parameter )
type       = token
subtype    = token

The type/subtype MAY be followed by parameters in the form of name=value pairs.
parameter  = token "=" ( token / quoted-string )
```

### Expected behaviour
When the content-type header is set to `text/html; charset="utf-8"` then the charset is parsed correctly as utf-8.

### Actual behaviour
When the content-type header is set to `text/html; charset="utf-8"` then the charset is parsed incorrectly as default ISO-8859-1.

### Steps to reproduce
Change the [`testGetCharset()`](https://github.com/netty/netty/blob/65e1f0e27c99c226041a21373985f584958ec971/codec-http/src/test/java/io/netty/handler/codec/http/HttpUtilTest.java#L78) to look like this:
```java
@Test
public void testGetCharset() {
    String NORMAL_CONTENT_TYPE = "text/html; charset=utf-8";
    String UPPER_CASE_NORMAL_CONTENT_TYPE = "TEXT/HTML; CHARSET=UTF-8";
    String CHARSET_IN_DOUBLE_QUOTES = "text/html; charset=\"utf-8\"";

    HttpMessage message = new DefaultHttpResponse(HttpVersion.HTTP_1_1, HttpResponseStatus.OK);
    message.headers().set(HttpHeaderNames.CONTENT_TYPE, NORMAL_CONTENT_TYPE);
    assertEquals(CharsetUtil.UTF_8, HttpUtil.getCharset(message));
    assertEquals(CharsetUtil.UTF_8, HttpUtil.getCharset(NORMAL_CONTENT_TYPE));

    message.headers().set(HttpHeaderNames.CONTENT_TYPE, UPPER_CASE_NORMAL_CONTENT_TYPE);
    assertEquals(CharsetUtil.UTF_8, HttpUtil.getCharset(message));
    assertEquals(CharsetUtil.UTF_8, HttpUtil.getCharset(UPPER_CASE_NORMAL_CONTENT_TYPE));

    message.headers().set(HttpHeaderNames.CONTENT_TYPE, CHARSET_IN_DOUBLE_QUOTES);
    assertEquals(CharsetUtil.UTF_8, HttpUtil.getCharset(message));
    assertEquals(CharsetUtil.UTF_8, HttpUtil.getCharset(CHARSET_IN_DOUBLE_QUOTES));
}
```

### Netty version
b0d1bff64

### PR DESCRIPTION
**Pull request by NiteshKant**
__Motivation__

As described in https://github.com/netty/netty/issues/11370 we should support quoted charset values

__Modification__

Modify `HttpUtil.getCharset(CharSequence contentTypeValue, Charset defaultCharset)` to trim the double-quotes if present.

__Result__
Fixes #11370:  `HttpUtil.getCharset()` now supports quoted charsets.

### CODE REVIEW

File: codec-http/src/test/java/io/netty/handler/codec/http/HttpUtilTest.java [hunk 1]
```diff
@@ -76,19 +76,66 @@ public void testGetCharsetAsRawCharSequence() {
...
+    }
+
+    @Test
+    public void testGetCharsetNoSpace() {
+        testGetCharsetUtf8("text/html;charset=utf-8");
+    }
+
+    @Test
+    public void testGetCharsetQuoted() {
+        HttpUtilTest.this.testGetCharsetUtf8("text/html; charset=\"utf-8\"");
```
Comment (kashike) [inline codec-http/src/test/java/io/netty/handler/codec/http/HttpUtilTest.java:89]: nit: unnecessary full-qualification here

File: codec-http/src/test/java/io/netty/handler/codec/http/HttpUtilTest.java [hunk 2]
```diff
@@ -76,19 +76,66 @@ public void testGetCharsetAsRawCharSequence() {
...
+    }
+
+    @Test
+    public void testGetCharsetNoSpaceQuoted() {
+        HttpUtilTest.this.testGetCharsetUtf8("text/html;charset=\"utf-8\"");
```
Comment (kashike) [inline codec-http/src/test/java/io/netty/handler/codec/http/HttpUtilTest.java:94]: nit: unnecessary full-qualification here

**Review by kashike — CHANGES_REQUESTED** [review]

File: codec-http/src/test/java/io/netty/handler/codec/http/HttpUtilTest.java [hunk 3]
```diff
@@ -76,19 +76,66 @@ public void testGetCharsetAsRawCharSequence() {
...
+    public void testGetCharsetNoTrailingQuotes() {
+        testGetCharsetInvalidQuotes("text/html;charset=\"utf-8");
+    }
+
+    @Test
+    public void testGetCharsetOnlyQuotes() {
+        testGetCharsetInvalidQuotes("text/html;charset=\"\"");
+    }
+
+    private void testGetCharsetInvalidQuotes(String contentType) {
```
Comment (normanmaurer) [inline codec-http/src/test/java/io/netty/handler/codec/http/HttpUtilTest.java:125]: nit: static

**Review by normanmaurer — CHANGES_REQUESTED** [review]

**Review by slandelle — APPROVED** [review]

**Review by normanmaurer — APPROVED** [review]

**Comment by normanmaurer** [comment]
@NiteshKant thanks a lot!

**Pull request merged**